### DISCUSSION TITLE
Ability to pass a disabled option to select/radio field

### DISCUSSION THREAD

**Discussion by ttbarnes** [keystonejs/keystone#7889] [Feature requests]
It'd be great to disable certain options passed to a select/radio field like so:

```js
status: select({
  options: [
    { label: 'A', value: 'A' },
    { label: 'B', value: 'B', isDisabled: true },
    { label: 'C', value: 'C' },
  ],
})
```

A use case would be for example, only certain user's with X roles can change the value to a special value, e.g i'd ultimately like to do this:

```js
status: select({
    options: [
      { label: 'A', value: 'A' },
      { label: 'B', value: 'B', isDisabled: !userHasPermission },
      { label: 'C', value: 'C' },
    ],
  })
```

Similar to the `access.operation` functionality, but on a deeper level.

It seems this ability might be possible by creating a custom field, feels overkill. Or another approach 🤔

**Comment by dcousens** [comment]
Related
- https://github.com/keystonejs/keystone/discussions/7424

My answer will echo the sentiments on that discussion.

There a number of complicating factors and architectural design decisions we are looking into for this feature insofar as it relates to conditional fields generally; sometimes the conditions/filters are dependent on the item context,  sometimes the conditions/filters are dependent on the targeted relationship and other times some kind of global context (for example, time).

We are considering these factors in ongoing work and you should hopefully expect an update on this this year.
In the interim,  you could look at rolling your own (as recommended by @fatihsolhan in https://github.com/keystonejs/keystone/discussions/7424).

Hope this helps :pray:

**Reply by ttbarnes** [reply]
Brilliant sounds great. Will look into the custom approach or alternative. Thank you @dcousens
